### ISSUE TITLE
Capture weather data from moth trap samples

### ISSUE THREAD

**Issue by DavidRoy** [BiologicalRecordsCentre/ABLE#499]
Need additional sample attributes:
- Moon phase
- start temperature
- end temperature
- minimum temperature (between start and end time, or midnight reading)
- wind speed (same categories as butterfly visits)

fields on data entry forms
look into using API service

**Comment by JurrienVanDeijk** [comment]
Would it be an idea to also capture the wind speed, cloud cover (in 8 parts), rainfall and humidity at the beginning and end of the night?

**Comment by CrisSevilleja** [comment]
Cloud cover I will keep it the same as for 15-min counts, percentage to don't mix different methods.

**Comment by DavidRoy** [comment]
Also need start and end time

**Comment by DavidRoy** [comment]
- [ ] survey setup
- [ ] website forms
- [ ] app forms

**Comment by Helder-Cardos** [comment]
On the website, under my Records, when editing a sample it does not allow to add moths species, only butterflies, from the drop-down list

**Comment by DavidRoy** [comment]
Thanks Helder - I've created a new issue for this as #506

**Comment by JimBacon** [comment]
Hi @DavidRoy 

Can you  provide some additional information, please.

- Moon phase. What terms do you want to use?
- Start/End/Minimum temperatures. Is this floating point, integer or term list? If the latter, what are the terms?
- % Cloud. Do you want this? At the beginning and end of the night?
- Rainfall & humidity. Do you want these? If so, what type of values do you wish to record? At the beginning and the end of the night?

**Comment by DavidRoy** [comment]
@Helder-Cardos @JurrienVanDeijk can you define the spec for this please

**Comment by Helder-Cardos** [comment]
@DavidRoy Apparently Jurrien is out of the office until the 13th Feb. It's better to wait for him to return and define this parameter together. I´ve mailed him already so we can post definitive parameters here.

**Comment by DavidRoy** [comment]
@Helder-Cardos @JurrienVanDeijk in general, I'm in favour of using other datasets to help with analysis.  It is best if we avoid asking too many questions of recorders.  Therefore, _could we just ask for start/end time for trap operation from recorders_ - then get weather, moon phase/intensity from existing sources?  This recent paper and R package looks useful?
https://link.springer.com/article/10.1007/s00265-022-03287-2.

In my experience, detailed weather data is rarely used in analysis and modellers often use summary data from existing sources.  Partly because asking recorders always means patchy data (if it's an optional element) whereas existing data (from models) is usually comprehensive (if less precise).

**Comment by Helder-Cardos** [comment]
@DavidRoy I agree with you.

 In fact, from our experience with the previous database we were using, not all recorders would fill with weather data all the time, and the resulting information was not very usable.

So, for me only the start and end time is enough, encouraging filling in the comments section if any disturbance occurred. Important is also the download data would include the start and end time, so we can calculate effort.

**Comment by chrisvanswaay** [comment]
I agree the weather details should not have to be filled in by the recorder. In fact the same could apply to start- and endtime. For monitoring we need to be sure the whole night was captured, from sunset to sunrise, and no doubt there is a place we can get those moments for each place on earth for each date if needed for an analysis. @JurrienVanDeijk should be back tomorrow, but if he agrees too, we could continue that way.

**Comment by Helder-Cardos** [comment]
@chrisvanswaay under our protocol in Portugal most sampling stations do sample all night, but we also allow for less sampling time, particularly using a light bulb and a sheet, where the minimum time is 3 hours after sunset. We have several moth stations that follow this protocol. So, having a start and ending time would be very useful for us

**Comment by DavidRoy** [comment]
We could set a default but editable start/end time based on sunset/rise?

**Comment by Helder-Cardos** [comment]
@DavidRoy that sounds great

**Comment by JurrienVanDeijk** [comment]
@Helder-Cardos , I am affraid I missed your email, I can't find it in my inbox (nor in the spam folder). But I agree on your comment @DavidRoy, to only ask the essential parts. In the Netherlands we only ask for the starting date, since we assume people only add data when they run a trap the whole night, But because they also use sheet data in Portugal it would be great to fill in a default start and endtime! But when there is also a possibility to add sheet data, wouldn't we need another trap type? or what type are you using now Helder?

**Comment by CrisSevilleja** [comment]
I think the weather data and time are captured on the app quite well and submitted to the database well. I think this issue can be closed.
